### PR TITLE
check relevant error returns

### PR DESCRIPTION
**Pull request by geyslan**
### 1. Explain what the PR does

Fix: #2477

commit 642144841699cdc95b4382bd77dfa6e08d774a3c

    builder: add staticcheck to dockerfiles

commit f511c8230f16d878eec3b6fc28a0e6e06afac361

    cleanup: check relevant return errors

commit 2bf32fc12466dabe1e28c087a9ca904d637f24a4

    tool: add errcheck
    
    Introduce `make check-err` rule and fix doc renaming `make code-check`
    to `make check-code`.

### 2. Explain how to test it

`make test-unit`
`sudo make test-integration`

### 3. Other comments

Context:

- #2555

Part of effort:

- #2477

### CODE REVIEW

**Comment by yanivagman** [comment]
I would like to review this before merge

**Comment by geyslan** [comment]
@yanivagman rebased and ready for review.

File: Makefile [hunk 1]
```diff
@@ -808,6 +809,20 @@ check-staticcheck: \
 		-tags $(GO_TAGS_EBPF) \
 		./...
 
+.PHONY: check-err
+check-err: \
+	.checkver_$(CMD_GO) \
+	tracee-ebpf \
```
Comment (yanivagman) [inline Makefile:815]: shouldn't we use tracee instead of tracee-ebpf?

**Review by yanivagman — APPROVED** [review]
Small clarification question, otherwise LGTM

File: Makefile (hunk 1 above)
Comment (geyslan) [reply Makefile:815]: Of course, but as you can see all other checks are still calling tracee-ebpf. I' assuming that when we get rid of tracee-ebpf and tracee-rules we will change all calls to them.

**Comment by geyslan** [comment]
This hasn't been merged yet as I'm not getting e2e tests to verify this PR. Check out run 685 and comments I put on its result.

**Pull request merged**